admin/tabs/activity/tracking/ActivityModel.js

```diff
@@ -0,0 +1,224 @@
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2020 Extremely Heavy Industries Inc.
+ */
+import {isEmpty, isFinite} from 'lodash';
+import {usernameCol} from '@xh/hoist/admin/columns';
+import {dateTimeCol, GridModel} from '@xh/hoist/cmp/grid';
+import {HoistModel, LoadSupport, managed, XH} from '@xh/hoist/core';
+import {fmtDate, numberRenderer} from '@xh/hoist/format';
+import {action, bindable, comparer, observable} from '@xh/hoist/mobx';
+import {LocalDate} from '@xh/hoist/utils/datetime';
+import {DimensionChooserModel} from '@xh/hoist/cmp/dimensionchooser';
+import {Cube} from '@xh/hoist/data';
+import {ChartsModel} from './charts/ChartsModel';
+
+@HoistModel
+@LoadSupport
+export class ActivityModel {
+
+    @bindable.ref startDate = LocalDate.today().subtract(7);
+    @bindable.ref endDate = LocalDate.today().add(1);  // https://github.com/xh/hoist-react/issues/400
+    @bindable username = '';
+    @bindable msg = '';
+    @bindable category = '';
+    @bindable device = '';
+    @bindable browser = '';
+
+    @observable.ref detailRecord = null;
+
+    // TODO: Create pref
+    @managed
+    dimChooserModel = new DimensionChooserModel({
+        enableClear: true,
+        dimensions: [
+            {label: 'Date', value: 'cubeDay'},
+            {label: 'User', value: 'username'},
+            {label: 'Message', value: 'msg'},
+            {label: 'Category', value: 'category'},
+            {label: 'Device', value: 'device'},
+            {label: 'Browser', value: 'browser'},
+            {label: 'User Agent', value: 'userAgent'}
+        ],
+        initialValue: ['cubeDay', 'username', 'msg']
+    });
+
+    @managed
+    cube = new Cube({
+        fields: [
+            {name: 'cubeDay', isDimension: true},
+            {name: 'username', isDimension: true, aggregator: 'UNIQUE'},
+            {name: 'msg', isDimension: true, aggregator: 'UNIQUE'},
+            {name: 'category', isDimension: true, aggregator: 'UNIQUE'},
+            {name: 'device', isDimension: true, aggregator: 'UNIQUE'},
+            {name: 'browser', isDimension: true, aggregator: 'UNIQUE'},
+            {name: 'userAgent', isDimension: true, aggregator: 'UNIQUE'},
+
+            {name: 'day', aggregator: 'RANGE'},
+            {name: 'elapsed', aggregator: 'AVG'},
+            {name: 'impersonating'},
+            {name: 'dateCreated'},
+            {name: 'data'},
+            {name: 'count', aggregator: 'CHILD_COUNT'},
+            {name: 'logCount', aggregator: 'LEAF_COUNT'}
+        ]
+    })
+
+    @managed
+    gridModel = new GridModel({
+        treeMode: true,
+        stateModel: 'xhActivityGrid',
+        enableColChooser: true,
+        enableExport: true,
+        exportOptions: {filename: () => `Activity ${fmtDate(this.startDate)} to ${fmtDate(this.endDate)}`},
+        emptyText: 'No activity reported...',
+        sortBy: 'dateCreated|desc',
```

Comment:
Seeing a warning in the console related to sorting, might be this line

Reply:
Seems to have gone away. Not sure what it might have been but can't repro any longer.